src/components/AddContactForm.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { validateName, validateImageUrl } from '../helpers/validation';
+import './AddContactForm.css';
+
+class AddContactForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      imageUrl: '',
+      firstNameError: false,
+      lastNameError: false,
+      imageUrlError: false
+    };
+  }
+
+  onFirstNameInputChange = (event) => {
+    const { target } = event;
+    const newFirstName = target.value.trim();
+    this.setState({ firstName: newFirstName });
+    if (newFirstName) {
+      this.setState({ firstNameError: !validateName(newFirstName) });
+    } else {
+      this.setState({ firstNameError: false });
+    }
+  }
+
+  onLastNameInputChange = (event) => {
+    const { target } = event;
+    const newLastName = target.value.trim();
+    this.setState({ lastName: newLastName });
+    if (newLastName) {
+      this.setState({ lastNameError: !validateName(newLastName) });
+    } else {
+      this.setState({ lastNameError: false });
+    }
+  }
+
+  onImageUrlInputChange = (event) => {
+    const { target } = event;
+    const newImageUrl = target.value.trim();
+    this.setState({ imageUrl: newImageUrl });
+    if (newImageUrl) {
+      this.setState({ imageUrlError: !validateImageUrl(newImageUrl) });
+    } else {
+      this.setState({ imageUrlError: false });
+    }
+  }
+
+  isButtonDisabled = () => {
+    const {
+      firstName,
+      lastName,
+      imageUrl,
+      firstNameError,
+      lastNameError,
+      imageUrlError
+    } = this.state;
+    return !(firstName && lastName && imageUrl) || firstNameError || lastNameError || imageUrlError;
+  }
+
+  handleSubmit = () => {
+    const { firstName, lastName, imageUrl } = this.state;
+    const { onSubmitAction } = this.props;
+    onSubmitAction(firstName, lastName, imageUrl);
+    this.setState({ firstName: '', lastName: '', imageUrl: '' });
+  }
+
+  render() {
+    const {
+      firstName,
+      lastName,
+      imageUrl,
+      firstNameError,
+      lastNameError,
+      imageUrlError
+    } = this.state;
+    return (
+      <div className="ui column grid">
+        <div className="thirteen wide column">
+          <div className="ui equal width form">
+            <h1 className="ui header center aligned">Add Contact</h1>
+            <div className="inline fields">
+              <div className={`wide field ${firstNameError && 'error'}`}>
```

Comment:
You should make a new component (something like <FormInput/>) and just provide the props it needs for each input. It should look like this:
```
<div className={`wide field ${firstNameError && 'error'}`}>
  <label>Name</label>
  <input
    className="text"
    placeholder="First Name"
    value={firstName}
    onChange={this.onFirstNameInputChange}
  />
</div>
```
where label, placeholder, type, etc values will come from props.